students/kyle_lehning/Lesson09/donor_models.py

```diff
@@ -0,0 +1,133 @@
+
+    @property
+    def avg_donation(self):
+        """Read only attribute that returns the average donation"""
+        try:
+            return round(self.total_donation / self.donation_num, 2)
+        except ZeroDivisionError:
+            return 0
+
+    def new_donation(self, value):
+        """
+        Increment the total_donation by the value passed.
+        Increment the donation_num by 1
+        Set most_recent_donation to be the passed value
+        """
+        self.total_donation += round(value, 2)
+        self.donation_num += 1
+        self.most_recent_donation = round(value, 2)
+
+    @classmethod
+    def from_existing(cls, don_name, don_total, don_number):
+        """
+        Allow a donor to be created that already includes its name, total value of donations, and number of donations.
+        This is useful if all this info already exists.
+        """
+        self = cls(don_name)
+        self.total_donation = don_total
+        self.donation_num = don_number
+        return self
+
+    def generate_recent_thanks(self):
+        """Return a string that is a thank you note based off the most_recent_donation value"""
+        string_to_format = ("Dear {},"
+                            "\n\nThank you for your kind donation of ${:.2f}."
+                            "\nIt will be put to very good use."
+                            "\n\nSincerely,"
+                            "\n-The Team")
+        return string_to_format.format(self.name, self.most_recent_donation)
+
+    def generate_total_thanks(self):
+        """Return a string that is a thank you note based off the total_donation value"""
+        string_to_format = ("Dear {},"
+                            "\n\nThank you for your kind donations totalling ${:.2f}."
+                            "\nIt will be put to very good use."
+                            "\n\nSincerely,"
+                            "\n-The Team")
+        return string_to_format.format(self.name, self.total_donation)
+
+
+class DonorCollection(object):
+    """
+    A class-based system for a collection of donors
+    """
+    def __init__(self, ):
+        """Initialize the object and make an empty list that will contain donor objects"""
+        self.donor_list = []
+
+    def __str__(self):
+        """returns a string saying what class is being used and the objects name"""
+        return (self.__class__.__name__ + " with {} donors").format(len(self.donor_list))
+
+    def add_new_donor(self, donor_name):
+        """Create a donor object based off the donor_name and add that object to the donor_list"""
+        self.donor_list.append(Donor(donor_name))
+
+    def add_existing_donor(self, don_name, don_total, don_number):
+        """Create a donor based off Donor's from_existing() function and add that donor to the donor_list"""
+        self.donor_list.append(Donor.from_existing(don_name, don_total, don_number))
```

Comment:
I don't really understand the purpose of this function here - in both cases you're adding a new donor object.